.github/workflows/CI_apptainer.yml

```diff
@@ -0,0 +1,37 @@
+name: Apptainer
+
+on:
+  push:
+    branches:
+      - 'master'
+    paths:
+      - '**'
+    tags:
+      - 'v*.*.*'
+  pull_request:
+    branches:
+      - 'master'
+    paths:
+      - '**'
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        arch: ['linux/amd64']
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: eWaterCycle/setup-apptainer@v2
+        with:
+            apptainer-version: 1.3.0
+      - name: Build apptainer
+        run: apptainer build pysr.sif Apptainer.def
+      - name: Test apptainer
+        run: apptainer exec ./pysr.sif /bin/bash -c 'pip install pytest nbval && PYTHONPATH=/pysr python3 -m pysr test main,cli,startup'
```

Comment:
docker has the `-v` flag for mounting a directory. Does `apptainer exec` have something like that, so that the tests have some way to write files to? It would just be for the tests (since some of the unittests write and load files)

Reply:
Also wouldn't the `pip install pytests nbval` also be writing to the image here?

Reply:
It looks like adding PYTHONPATH=/pysr obviates the need to switch to the /pysr directory. If the directory the tests are being run from is writable adding an additional mount point should be unnecessary.